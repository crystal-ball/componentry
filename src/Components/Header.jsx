import React from 'react'
import { string } from 'prop-types'

Header.propTypes = {
  title: string
}

Header.defaultProps = {
  title: 'Components'
}

export default function Header({ title }) {
  return (
    <div className="jumbotron jumbotron-fluid bg-primary">
      <div className="container">
        <h1 className="display-3 text-light">
          {title.charAt(0).toUpperCase() + title.slice(1)}
        </h1>
      </div>
    </div>
  )
}
