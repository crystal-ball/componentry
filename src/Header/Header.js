// @flow
import React, { type Node } from 'react'
import elementFactory from '../component-factories/element'

type Props = {
  children: Node,
  /** Activates subheader styles, including reduction in font size and muted text color */
  subheader?: boolean,
}

export default elementFactory(
  'Header',
  ({ subheader, children, ...props }: Props) => ({
    as: 'h1',
    className: {
      'text-muted': subheader,
    },
    children: subheader ? <small>{children}</small> : children,
    ...props,
  }),
)
