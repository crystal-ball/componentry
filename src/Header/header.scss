/* componentry/src/Header/header */

$heading-sizes: (
  h1: $h1-font-size,
  h2: $h2-font-size,
  h3: $h3-font-size,
  h4: $h4-font-size,
  h5: $h5-font-size,
  h6: $h6-font-size,
);

$heading-weights: (
  h1: $h1-font-weight,
  h2: $h2-font-weight,
  h3: $h3-font-weight,
  h4: $h4-font-weight,
  h5: $h5-font-weight,
  h6: $h6-font-weight,
);

// Creating individual output for each header adds ~25 lines of CSS, but also
// makes debugging styles much cleaner, and more closely aligns with the goal of
// mapping classes to usage 1:1.
@each $heading, $heading-font-size in $heading-sizes {
  #{$heading},
  .#{$heading} {
    margin-top: 0;
    margin-bottom: $headings-margin-bottom;
    font-family: $headings-font-family;
    font-style: $headings-font-style;
    font-weight: map-get($map: $heading-weights, $key: $heading);
    line-height: $headings-line-height;
    color: $headings-color;
    font-size: $heading-font-size;
  }
}
