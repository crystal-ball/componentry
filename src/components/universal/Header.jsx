import React from 'react'

type Props = {
  title: string
}

export default ({ title = 'Components' }: Props) => (
  <div className="jumbotron jumbotron-fluid bg-secondary">
    <div className="container">
      <h1 className="display-3 text-light">
        {title.charAt(0).toUpperCase() + title.slice(1)}
      </h1>
    </div>
  </div>
)
