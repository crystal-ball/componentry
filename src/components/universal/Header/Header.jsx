import React from 'react'

type Props = {
  title: string,
}

export default ({ title }: Props) => (
  <div className="mb-5 bb-secondary-1">
    <h1 className="display-3 mb-5">
      {title.charAt(0).toUpperCase() + title.slice(1)}
    </h1>
  </div>
)
