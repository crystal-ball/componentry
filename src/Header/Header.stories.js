/* global module */

import React from 'react'
import { storiesOf } from '@storybook/react'

import Header from './Header'

storiesOf('Header', module).add('<Header />', () => <Header>Componentry header</Header>)
