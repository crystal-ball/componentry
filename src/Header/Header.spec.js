import React from 'react'
import { render } from '@testing-library/react'

import Header from './Header'
import elementTests from '../../test/element-tests'

describe('<Header/>', () => {
  // Basic library element test suite
  elementTests(Header)
})

// Snapshots
// ---------------------------------------------------------------------------
describe('<Header /> snapshots', () => {
  it('renders correctly', () => {
    const { container } = render(<Header>Header</Header>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
