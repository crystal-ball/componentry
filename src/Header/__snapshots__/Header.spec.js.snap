// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshots renders correctly 1`] = `
<h1
  class="h1"
>
  Header
</h1>
`;
